db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java

```diff
@@ -82,23 +135,166 @@ public static String toString(PredicateList predicateList) {
+    private static void throwNotImplementedError() throws StandardException {
+        throw StandardException.newException(SQLState.LANG_DOES_NOT_IMPLEMENT);
+    }
+
+    // We don't support REAL (float), because the way spark
+    // evaluates expressions involving float causes accuracy errors
+    // that don't occur when splice does the evaluation.
+    private static boolean sparkSupportedType(int typeFormatId) {
+        return (typeFormatId == BOOLEAN_TYPE_ID  ||
+                typeFormatId == DATE_TYPE_ID     ||
+                typeFormatId == CHAR_TYPE_ID     ||
+                typeFormatId == VARCHAR_TYPE_ID  ||
+                typeFormatId == LONGVARCHAR_TYPE_ID  ||
+                typeFormatId == TINYINT_TYPE_ID  ||
+                typeFormatId == SMALLINT_TYPE_ID ||
+                typeFormatId == INT_TYPE_ID      ||
+                typeFormatId == LONGINT_TYPE_ID  ||
+                typeFormatId == DECIMAL_TYPE_ID  ||
+                typeFormatId == DOUBLE_TYPE_ID   ||
+                typeFormatId == TIMESTAMP_TYPE_ID);
+    }
+
+    private static boolean isNumericTypeFormatID(int typeFormatId) {
+        return (typeFormatId == TINYINT_TYPE_ID  ||
+                typeFormatId == SMALLINT_TYPE_ID ||
+                typeFormatId == INT_TYPE_ID      ||
+                typeFormatId == LONGINT_TYPE_ID  ||
+                typeFormatId == DECIMAL_TYPE_ID  ||
+                typeFormatId == DOUBLE_TYPE_ID   ||
+                typeFormatId == REAL_TYPE_ID);
+    }
+
+    private static boolean isOverflowSensitive(ValueNode operand) throws StandardException {
+        return (operand.getTypeId().getTypeFormatId() == LONGINT_TYPE_ID ||
+                operand.getTypeId().getTypeFormatId() == DECIMAL_TYPE_ID ||
+                operand.getTypeId().getTypeFormatId() == DOUBLE_TYPE_ID);
+    }
+    private static void checkOverflowHidingCases(BinaryArithmeticOperatorNode bao) throws StandardException {
+        if (RELATIONAL_OP_DEPTH.get().intValue() <= 0)
+            return;
+        ValueNode leftOperand = bao.getLeftOperand();
+        ValueNode rightOperand = bao.getRightOperand();
+        if (isOverflowSensitive(leftOperand) || isOverflowSensitive(rightOperand))
+            throwNotImplementedError();
+    }
+
+
+    private static String opToString2(ValueNode operand) throws StandardException {
         if(operand==null){
             return "";
-        }else if(operand instanceof UnaryOperatorNode){
+        } else if (operand instanceof UntypedNullConstantNode)
+            return " null ";
+        else if(operand instanceof UnaryOperatorNode){
             UnaryOperatorNode uop=(UnaryOperatorNode)operand;
-            return format("%s(%s)",uop.getOperatorString(),opToString(uop.getOperand()));
+            String operatorString = uop.getOperatorString();
+            if (SPARK_EXPRESSION.get()) {
+                if (operand instanceof IsNullNode) {
+                    RELATIONAL_OP_DEPTH.get().increment();
```

Comment:
I think we should use a normal function parameter rather than a ThreadLocal here too.

Reply:
Removed all ThreadLocals in OperatorToString.